### PR TITLE
removed _ from struct names in core folder

### PR DESCRIPTION
**Pull request by AlexHolly**
part of #56 for 3.0

Contains alle struct renamings in the core foder except for "Variant"

### CODE REVIEW

File: core/bind/core_bind.cpp [hunk 1]
```diff
@@ -842,7 +842,7 @@ void _OS::print_all_textures_by_size() {
 			Size2 size = E->get()->call("get_size");
 			int fmt = E->get()->call("get_format");
 
-			_OSCoreBindImg img;
+            OSCoreBindImg img;
```
Comment (akien-mga) [inline core/bind/core_bind.cpp]: Looks like your IDE is configured to force spaces for indentation :(

**Comment by volzhs** [comment]
I guess @akien-mga doesn't like 4 spaces instead of tab. :)

**Comment by reduz** [comment]
Just removing the underscore I think is not enough, I would add Private or
Internal at the end of the name, given this is meant to be private

File: core/typedefs.h [hunk 2]
```diff
@@ -273,13 +273,13 @@ struct Comparator {
...
+struct GlobalLock {
 
-	_GlobalLock() { _global_lock(); }
-	~_GlobalLock() { _global_unlock(); }
+    GlobalLock() { _global_lock(); }
+    ~GlobalLock() { _global_unlock(); }
 };
 
-#define GLOBAL_LOCK_FUNCTION _GlobalLock _global_lock_;
+#define GLOBAL_LOCK_FUNCTION GlobalLock _global_lock_;
```
Comment (akien-mga) [inline core/typedefs.h]: Windows/MinGW seems not to like the change: https://travis-ci.org/godotengine/godot/jobs/146396830

**Comment by AlexHolly** [comment]
OFF TOPIC
@akien-mga

Can you add the indentation settings to the docs?
http://docs.godotengine.org/en/latest/reference/configuring_an_ide.html

Qt Creator 3.0.1 based on Qt 5.2.1
Projects -> Code Style -> Edit... -> Copy Built-in Code Style -> (General) Tap policy -> "Tabs Only"

(Would also be nice to see the version of each ide)

**Comment by akien-mga** [comment]
@AlexHolly Pull requests welcome :) https://github.com/godotengine/godot-docs/blob/master/reference/configuring_an_ide.rst

**Comment by akien-mga** [comment]
Closing as it does not build, and as per:
> Just removing the underscore I think is not enough, I would add Private or Internal at the end of the name, given this is meant to be private

**Pull request closed without merging**